docs/syntaxes.md

```diff
@@ -0,0 +1,64 @@
+# Syntaxes
+
+Melange supports two syntaxes: OCaml syntax (the default) and Reason syntax.
+Choice of syntax doesn't change the output of the compiler, but Reason syntax is
+designed to be more approachable to JavaScript/TypeScript developers, and it
+supports the use of JSX. Most tooling, such as the [OCaml Platform VS Code
+extension](https://marketplace.visualstudio.com/items?itemName=ocamllabs.ocaml-platform),
+works well with Reason, but some tools aren't completely aware of it—for
+example, error messages from the compiler still use OCaml syntax.
+
+## Installation
+
+To install Reason syntax support add `reason` to the `depends` section of the
+`<project-name>.opam` file in your project:
+
+```opam{3}
+depends: [
+  "ocaml"
+  "reason" {>= "3.10.0"}
+  "dune" {>= "3.9"}
+  "melange" {>= "2.0.0"}
+  "reason-react" {>= "0.13.0"}
+  "reason-react-ppx"
+  "opam-check-npm-deps" {with-test}
+  "ocaml-lsp-server" {with-test}
+  "dot-merlin-reader" {with-test}
+  "odoc" {with-doc}
+]
+```
+
+On the command line, run
+
+```bash
+opam install -y . --deps-only
+```
+
+Note that reason support is already set up for you in
+[melange-opam-template](https://github.com/melange-re/melange-opam-template).
+
+## Editor configuration
+
+To enable format-on-save in VS Code:
```

Comment:
Probably link to reasonml editor for more info? and same for OCaml?

Reply:
What page do you want to link to?

Reply:
Probably https://reasonml.github.io/docs/en/editor-plugins

Reply:
Added link in https://github.com/melange-re/melange-re.github.io/pull/200/commits/c36a87ef4bf38a8dae5c0fe596b5ade0d3139457